server/routes/utilMongoose.js

```diff
@@ -359,7 +359,14 @@ router.get('/allRequests', async (req, res, next) => {
 
 router.get('/populateAllRequests', async (req, res, next) => {
     try {
-        const dbData = await requestSchema.find().populate('alumni').populate('student')
+        const dbData = await requestSchema.find().populate('mentor')
+        for (let request of dbData) {
+            if (request.requesterRole === 'STUDENT') {
+                request.requesterObj = await studentSchema.findOne({_id: request.requester})
+            } else {
+                request.requesterObj = await alumniSchema.findOne({_id: request.requester})
+            }
+        }
```

Comment:
We can no longer use a cursor to automatically populate requester profiles because we don't know which collection to reference
This is a workaround. We can still populate the mentor, but for each request we need to determine if they are a student or an alumni, find the correct record, and then return the object.
We can't do this in request.requester, because it will be converted to a string. Instead, I've added the requesterObj as a field used to return the student/alumni object since you can't turn requester into an object